### PR TITLE
Login flow refactor

### PR DESCRIPTION
**Pull request by lemieux**
This is a simple one that tackles the login flow, which doesn't do much aside setting the right props in the store.

### CODE REVIEW

File: src/frame/js/actions/auth.js [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
-export function login(props) {
+export function login() {
     return (dispatch, getState) => {
-        const {config: {appId}} = getState();
-        return dispatch(http('POST', `/apps/${appId}/login`, props));
+        const {config: {appId}, user: {userId}, auth: {sessionToken}} = getState();
+        return dispatch(http('POST', `/apps/${appId}/login`, {
+            userId,
+            sessionToken,
+            clientId: getDeviceId(appId)
```
Comment (mspensieri) [inline src/frame/js/actions/auth.js]: Isn't login accepting the whole `client` object?

File: src/frame/js/actions/conversation.js [hunk 2]
```diff
@@ -515,18 +517,31 @@ export function disconnectFaye() {
...
 
+        if (sessionToken) {
+            actions.push(setAuth({
+                sessionToken
+            }));
+        }
+
         dispatch(batchActions(actions));
+
+        if (appUser.conversationStarted) {
```
Comment (mspensieri) [inline src/frame/js/actions/conversation.js]: might be better to check `if(conversation)` since we have it?

File: src/frame/js/utils/device.js [hunk 3]
```diff
@@ -2,8 +2,8 @@ import uuid from 'uuid';
 
 import storage from './storage';
 
-export function getDeviceId() {
-    const SK_STORAGE = 'sk_deviceid';
+export function getDeviceId(appId) {
+    const SK_STORAGE = `${appId}.clientId`;
```
Comment (mspensieri) [inline src/frame/js/utils/device.js]: How come client id is scoped per app now?

File: src/frame/js/actions/auth.js (hunk 1 above)
Comment (lemieux) [inline src/frame/js/actions/auth.js]: @hebime told me only clientId was fine.

File: src/frame/js/actions/conversation.js (hunk 2 above)
Comment (lemieux) [inline src/frame/js/actions/conversation.js]: I guess we could, but that'd be a backend logic error if it returns `conversationStarted` for the user and not return the conversation with it.

File: src/frame/js/actions/auth.js (hunk 1 above)
Comment (wmora) [inline src/frame/js/actions/auth.js]: but isn't it `client.id`?

Comment (hebime) [inline src/frame/js/actions/auth.js]: we take the entire client now 😞

File: src/frame/js/actions/conversation.js [hunk 4]
```diff
@@ -90,23 +91,23 @@ const throttlePerUser = (userId) => {
 
 function postSendMessage(message) {
     return (dispatch, getState) => {
-        const {user: {_id}} = getState();
+        const {config: {appId}, user: {_id}} = getState();
         return dispatch(http('POST', `/appusers/${_id}/messages`, {
             ...message,
             role: 'appUser',
-            deviceId: getDeviceId()
+            deviceId: getDeviceId(appId)
```
Comment (wmora) [inline src/frame/js/actions/conversation.js]: (for future PR) this goes in a separate `sender.client` object now

**Comment by wmora** [comment]
late 👍

**Pull request merged**